stripe/src/test/java/com/stripe/android/testharness/JsonTestUtils.java

```diff
@@ -0,0 +1,179 @@
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Utility class for testing with JSON objects and maps that are created from or for JSON.
+ */
+public class JsonTestUtils {
+
+    /**
+     * Assert that two {@link JSONObject JSONObjects} are equal, comparing key by key recursively.
+     *
+     * @param first the first object
+     * @param second the second object
+     */
+    public static void assertJsonEquals(JSONObject first, JSONObject second) {
+        if (assertSameNullity(first, second)) {
+            return;
+        }
+
+        assertEquals(first.length(), second.length());
+        Iterator<String> keyIterator = first.keys();
+        while(keyIterator.hasNext()) {
+            String key = keyIterator.next();
+            assertTrue(second.has(key));
+            if (first.opt(key) instanceof JSONObject) {
+                assertTrue(second.opt(key) instanceof JSONObject);
+                assertJsonEquals(first.optJSONObject(key), second.optJSONObject(key));
+            } else if (first.opt(key) instanceof JSONArray) {
+                assertTrue(second.opt(key) instanceof JSONArray);
+                assertJsonArrayEquals(first.optJSONArray(key), second.optJSONArray(key));
+            } else if (first.opt(key) instanceof Long || first.opt(key) instanceof Integer) {
+                assertTrue(second.opt(key) instanceof Long ||
+                        second.opt(key) instanceof Integer);
+                assertNumericFuzzyEquals(first.opt(key), second.opt(key));
+            } else {
+                assertEquals(first.opt(key), second.opt(key));
+            }
+        }
+    }
+
+    /**
+     * Assert that two {@link JSONArray JSONArrays} are equal, comparing index by index recursively.
+     *
+     * @param first the first array
+     * @param second the second array
+     */
+    public static void assertJsonArrayEquals(JSONArray first, JSONArray second) {
+        if (assertSameNullity(first, second)) {
+            return;
+        }
+        assertEquals(first.length(), second.length());
+        for (int i = 0; i < first.length(); i++) {
+            if (first.opt(i) instanceof JSONObject) {
+                assertTrue(second.opt(i) instanceof JSONObject);
+                assertJsonEquals(first.optJSONObject(i), second.optJSONObject(i));
+            } else if (first.opt(i) instanceof JSONArray) {
+                assertTrue(second.opt(i) instanceof JSONArray);
+                assertJsonArrayEquals(first.optJSONArray(i), second.optJSONArray(i));
+            } else if (first.opt(i) instanceof Long || first.opt(i) instanceof Integer) {
+                assertTrue(second.opt(i) instanceof Long ||
+                        second.opt(i) instanceof Integer);
+                assertNumericFuzzyEquals(first.opt(i), second.opt(i));
```

Comment:
Aha! That's the function I was looking for. I just wanted to coerce a numeric object type to a long primitive.